internal/host/logscmd.go

```diff
@@ -0,0 +1,46 @@
+package host
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-service/models"
+)
+
+type logsCmd struct {
+	baseCmd
+	instructionConfig InstructionConfig
+}
+
+func NewLogsCmd(log logrus.FieldLogger, instructionConfig InstructionConfig) *logsCmd {
+	return &logsCmd{
+		baseCmd:           baseCmd{log: log},
+		instructionConfig: instructionConfig,
+	}
+}
+
+func (i *logsCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models.Step, error) {
+	// added to run upload logs if install command fails
+	if host.LogsCollectedAt != strfmt.DateTime(time.Time{}) {
+		return nil, nil
+	}
+	logsCommand, err := CreateUploadLogsCmd(host, i.instructionConfig.ServiceBaseURL,
+		i.instructionConfig.InventoryImage, i.instructionConfig.SkipCertVerification,
+		false, true)
```

Comment:
Don't we want to parameterise the `withInstallerGatherLogs` here? You assume we always run installer-gather (which is true in assisted-service) but if we already made it a separate command maybe we want to make it flexible no?!